src/bgw/job_stat.c

```diff
@@ -238,6 +238,7 @@ calculate_next_start_on_failure(TimestampTz finish_time, int consecutive_failure
 	}
 	PG_CATCH();
 	{
+		MemoryContextSwitchTo(oldctx);
```

Comment:
Not sure how this is related to removing job_type. It might be necessary to add this line because of a bug, but I have no way of understanding why this line is necessary since I have no context. Either put it in a separate commit that explains why it was added, or add some sort of comment here (I would prefer the former).

Reply:
moved into separate PR